test/framework/prometheus.go

```diff
@@ -319,17 +321,14 @@ func (f *Framework) CreatePrometheusAndWaitUntilReady(ctx context.Context, ns st
 	return result, nil
 }
 
-func (f *Framework) ScalePrometheusAndWaitUntilReady(ctx context.Context, name, ns string, replicas int32) (*monitoringv1.Prometheus, error) {
-	return f.PatchPrometheusAndWaitUntilReady(
-		ctx,
-		name,
-		ns,
-		monitoringv1.PrometheusSpec{
-			CommonPrometheusFields: monitoringv1.CommonPrometheusFields{
-				Replicas: ptr.To(int32(replicas)),
-			},
-		},
-	)
+func (f *Framework) ScalePrometheusAndWaitUntilReady(ctx context.Context, p *monitoringv1.Prometheus, ns string, shards int32) (*monitoringv1.Prometheus, error) {
```

Comment:
(nit) could use p.Namespace directly?
```suggestion
func (f *Framework) ScalePrometheusAndWaitUntilReady(ctx context.Context, p *monitoringv1.Prometheus, shards int32) (*monitoringv1.Prometheus, error) {
```